plugins/micromega/persistent_cache.ml

```diff
@@ -130,7 +130,30 @@ module PHashtable (Key : HashedType) : PHashtable with type key = Key.t = struct
   let do_under_lock kd fd f =
     if lock kd fd then fun_protect f ~finally:(fun () -> unlock fd) else f ()
 
-  let fopen_in = open_in
+  let fopen_in = open_in_bin
+
+  (* from lib/objFile.ml *)
+  let input_int32 ch =
+    let accu = ref 0l in
+    for _i = 0 to 3 do
+      let c = input_byte ch in
+      accu := Int32.add (Int32.shift_left !accu 8) (Int32.of_int c)
+    done;
+    !accu
+
+  let output_int32 ch n =
+    for i = 0 to 3 do
+      output_byte ch (Int32.to_int (Int32.shift_right_logical n (24 - 8 * i)))
+    done
```

Comment:
It would be better to just export these functions from `ObjFile`.

Reply:
I agree that duplication is not the right way. Yet it seems `ObjFile` does not expose this kind of stuff (on purpose?).
Could `lib/system.mli` be the home for these functions?

Reply:
Either that or `lib/util.mli`. No strong opinion either way.